ldap/servers/slapd/back-ldbm/db-mdb/mdb_layer.c

```diff
@@ -2116,13 +2112,24 @@ void *dbmdb_recno_cache_build(void *arg)
         dbmdb_generate_recno_cache_key_by_recno(&rckey, recno);
         rc = MDB_PUT(txn_ctx.txn, rcctx->rcdbi->dbi, &rckey, &rcdata, 0);
         slapi_ch_free(&rckey.mv_data);
-        if (rc == 0) {
+        if (rc) {
+            slapi_log_err(SLAPI_LOG_ERR, "dbmdb_recno_cache_build", 
+                          "Failed to write record in db %s, key=%s error: %s\n", 
+                            rcctx->rcdbi->dbname, (char*)(key.mv_data), mdb_strerror(rc));
+        } else {
             dbmdb_generate_recno_cache_key_by_data(&rckey, &key, &data);
             rc = MDB_PUT(txn_ctx.txn, rcctx->rcdbi->dbi, &rckey, &rcdata, 0);
             slapi_ch_free(&rckey.mv_data);
             txn_ctx.flags |= DBMDB_TXNCTX_NEED_COMMIT;
+            if (rc) {
+                slapi_log_err(SLAPI_LOG_ERR, "dbmdb_recno_cache_build", 
+                              "Failed to write record in db %s, key=%s error: %s\n", 
+                                rcctx->rcdbi->dbname, (char*)(key.mv_data), mdb_strerror(rc));
+            }
         }
         slapi_ch_free(&rcdata.mv_data);
+        if (rc) { 
+        }
```

Comment:
Is this empty block intentional?

Reply:
No. I wanted to add to check on code return but that is not even the right place ... Fixing that ...